lib/package.gi

```diff
@@ -1862,19 +1862,19 @@ InstallGlobalFunction( SetPackagePath, function( pkgname, pkgpath )
 #F  ExtendRootDirectories( <paths> )
 ##
 InstallGlobalFunction( ExtendRootDirectories, function( rootpaths )
-    local i;
-
-    rootpaths:= Filtered( rootpaths, path -> not path in GAPInfo.RootPaths );
-    if not IsEmpty( rootpaths ) then
-      # 'DirectoriesLibrary' concatenates root paths with directory names.
-      for i in [ 1 .. Length( rootpaths ) ] do
-        if not EndsWith( rootpaths[i], "/" ) then
-          rootpaths[i]:= Concatenation( rootpaths[i], "/" );
-        fi;
-      od;
-      # Append the new root paths.
-      GAPInfo.RootPaths:= Immutable( Concatenation( GAPInfo.RootPaths,
```

Comment:
I've replaced this now with a loop that uses `Add` to add each path separately. Various reasons
- the "skip paths already in the list" is more powerful if we do it after ensuring the trailing slash is there
- for HPC-GAP compatibility we may have to switch `GAPInfo.RootPaths` from a plist to an atomic list, where `Add` is supported but not `Append`

Anyway this is of course also a natural place to turn relative paths in absolute ones, call `realpath`, etc. -- but that goes beyond the scope of this PR